docs/source/api.rst

```diff
@@ -125,3 +125,67 @@ For example::
 
 
 
+Detailed API reference
+------------------------
+Module: :mod:`traitlets.traitlets`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. automodule:: traitlets.traitlets
+    :members:
+    :private-members:
+
+Module: :mod:`traitlets.config.configurable`
```

Comment:
In fact, I think this is generally a better approach - to carefully whitelist what APIs are considered public and documented. Liberal `automodule` has caused us many problems in the IPython project, and is specifically why we are trying something else on newer projects like this one. Each public API should be carefully and deliberately considered before committing it to the public API documentation.